src/main/java/org/candlepin/resteasy/filter/VerifyAuthorizationFilter.java

```diff
@@ -251,40 +254,39 @@ protected List<Persisted> getAccessedEntities(Verify verify, Object requestValue
         }
 
         List<Persisted> entities = new ArrayList<>();
-        Class<? extends Persisted> verifyType = verify.value();
-
-        if (requestValue instanceof String) {
-            String verifyParam = (String) requestValue;
-            Persisted entity = null;
-
-            entity = storeFactory.getFor(verifyType).lookup(verifyParam);
-
-            // If the request is just for a single item, throw an exception
-            // if it is not found.
-            if (entity == null) {
-                // This is bad, we're verifying a parameter with an ID which
-                // doesn't seem to exist in the DB. Error will be thrown in
-                // invoke though.
-                String typeName = Util.getClassName(verifyType);
-                if (typeName.equals("Owner")) {
-                    typeName = i18nProvider.get().tr("Organization");
+        Class<? extends Persisted>[] verifyTypes = verify.value();
+
+        for (Class<? extends Persisted> verifyType : verifyTypes) {
+            if (requestValue instanceof String) {
+                String verifyParam = (String) requestValue;
+                Persisted entity = storeFactory.getFor(verifyType).lookup(verifyParam);
+                if (entity == null) {
+                    String typeName = Util.getClassName(verifyType);
+                    if (typeName.equals("Owner")) {
+                        typeName = i18nProvider.get().tr("Organization");
+                    }
+
+                    log.info("No such entity: {}, id: {}", typeName, verifyParam);
+                    continue;
                 }
-                String msg = i18nProvider.get().tr("{0} with id {1} could not be found.",
-                    typeName, verifyParam);
-                log.info("No such entity: {}, id: {}", typeName, verifyParam);
-                throw new NotFoundException(msg);
+
+                entities.add(entity);
             }
+            else if (requestValue instanceof Collection) {
+                Collection<String> verifyParams = (Collection<String>) requestValue;
 
-            entities.add(entity);
+                // If the request is for a list of items, we'll leave it
+                // up to the requester to determine if something is missing or not.
+                if (verifyParams != null && !verifyParams.isEmpty()) {
```

Comment:
You can just check for emptiness, `verifyParams` cannot be null.

Reply:
Updated